src/assets/js/protocol/protocol-details.js

```diff
@@ -0,0 +1,229 @@
+            return;
+        }
+
+        // add class to parent to indicate js initialized
+        parent.classList.add('is-details');
+
+        // Expand
+        // siblings of the summary, until next summary
+        summarySiblings = window.Mozilla.Utils.nextUntil(summary, selector);
+
+        // look to see if all children are already in a wrapper we can use
+        if(summarySiblings.length == 1) {
+            details = summarySiblings[0];
+        } else if (summarySiblings.length > 1){
+            details = doc.createElement('div');
+            details.classList.add('mzp-c-details-wrapper');
+            summarySiblings.forEach(function(sibling) {
+                details.append(sibling);
+            });
+
+            summary.parentNode.insertBefore(details, summary.nextSibling);
+        } else {
+            // error
+            // TODO: what do we do with errors? fail silently?
+        }
+
+        // look for existing ID to use
+        if(!details.id) {
+            // if details already has ID, use that, if not assign one
+            var unique = selector.replace(/[^a-zA-Z]+/g, '');
+            details.id = 'expand-' + unique + '-'+ _count;
+            _count++;
+        }
+
+        // close by default
+        // TODO: add support for open attribute
+        details.setAttribute('aria-hidden', true);
+        details.classList.add('is-closed');
+
+        // Control
+        control.setAttribute('type', 'button');
+        // add aria-controls
+        control.setAttribute('aria-controls', details.id);
+        // add aria-expanded
+        control.setAttribute('aria-expanded', false);
+        // add listener
+        control.addEventListener('click', Details.handleControlActivation, false);
+        // copy the summary's contents into the control
+        summary.childNodes.forEach(function(child) {
+            control.append(child);
+        });
+        // append control element
+        summary.appendChild(control);
+        summary.classList.add('is-summary');
+    };
+
+    /**
+     * destroyItem
+     * @param {Object} el - Element the control was placed inside of
+     * - does not attempt to remove the details wrapper
+     */
+    Details.destroyItem = function(el) {
+        var summary = el;
+        var parent = summary.parentNode;
+        var details = summary.nextElementSibling;
+        var control = summary.querySelector('button');
+        parent.classList.remove('is-details');
+        details.removeAttribute('aria-hidden');
+        details.classList.remove('is-closed');
+        // move control's contents back to summary
+        control.childNodes.forEach(function(child) {
+            summary.append(child);
+        });
+        summary.removeChild(control);
+        summary.classList.remove('is-summary');
+    };
+
+    /**
+     * Init
+     * @param {Object} selector - CSS selector matching "summary" elements
+     */
+    Details.init = function(selector) {
```

Comment:
Since this code path depends on `Mozilla.Utils.nextUntil`, should we add a check that it is defined so it fails gracefully if not?

Reply:
Added a check 👍